src/data/nat/basic.lean

```diff
@@ -449,6 +449,13 @@ rw [mul_assoc, nat.add_mul_div_left _ _ (nat.pos_of_ne_zero hb), add_mul_mod_sel
 lemma mod_mul_left_div_self (a b c : ℕ) : a % (c * b) / b = (a / b) % c :=
 by rw [mul_comm c, mod_mul_right_div_self]
 
+/- The `n+1`-st triangle number is `n` more than the `n`-th triangle number -/
+lemma triangle_succ (n : ℕ) : (n + 1) * ((n + 1) - 1) / 2 = n * (n - 1) / 2 + n :=
```

Comment:
Do we want more stuff about triangle numbers? Or should we just use `choose n 2`? (That would give this lemma quite easily.) Maybe we want a lemma that computes `choose n 2 = (n+1) * n / 2`?

Reply:
Why not work in rat? Then the division will be far easier to deal with.

Reply:
I'm getting the triangle numbers because of
```lean
lemma sum_range_id (n : ℕ) : (finset.range n).sum (λi, i) = (n * (n - 1)) / 2
```
If our preferred way is to write triangle numbers as `choose n 2`, then that lemma should also be changed.

Reply:
I proved that `choose n 2 = (n+1) * n / 2` (using this lemma)

Reply:
@kbuzzard: because I want a natural number in the end, and then I'd have to go from `rat` to `nat` which gives me extra boiler plate.